src/station_gui.cpp

```diff
@@ -3096,3 +3096,51 @@ bool ShouldShowBaseStationViewportLabel(const BaseStation *bst)
 			!HasBit(_extra_display_opt, XDO_SHOW_HIDDEN_SIGNS)) return false;
 	return true;
 }
+
+enum StationTooltipNameMode : uint8 {
+	STNM_OFF,
+	STNM_ON_IF_HIDDEN,
+	STNM_ALWAYS_ON
+};
+
+char *StationGetSpecialStringExternal(char *buff, int x, const char *last);
+
+bool GetStationViewportTooltipString(const TileIndex tile, char *buffer_position, const char *buffer_tail)
+{
+	const StationID station_id = GetStationIndex(tile);
+	const Station *station = Station::Get(station_id);
+
+	bool next = false;
+
+	if (_settings_client.gui.station_viewport_tooltip_name == STNM_ALWAYS_ON ||
```

Comment:
When mixing && and || it's a good idea to use parentheses. Currently it is ambiguous what you mean with these.